server/datastore/mysql/apple_mdm.go

```diff
@@ -953,8 +953,15 @@ func (ds *Datastore) ListMDMAppleProfilesToInstall(ctx context.Context) ([]*flee
           WHERE
           hmap.profile_id IS NULL
           AND hmap.host_uuid IS NULL
-          AND hmap.status != 'pending' OR hmap.status IS NULL
-          AND hmap.operation_type != 'install' OR hmap.operation_type IS NULL
+					-- TODO(mna): correct me if I'm wrong but if hmap.profile_id and hmap.host_uuid
+					-- must be NULL, and those are the non-nullable primary keys of hmap, then any
+					-- other hmap conditions don't matter - we care about profiles that exist in macp but
+					-- not at all in hmap? (i.e. it's impossible for hmap.profile_id to be NULL AND for
+					-- hmap.status to be anything but NULL, same for hmap.operation_type).
+					-- I think what we want here is actually not just profile_id+host_uuid being NULL,
+					-- but EITHER NULL OR NOT NULL and the status/operation conditions?
```

Comment:
@mna you're right, with the current logic, the last two `AND`s are redundant, we only care about the rows in `mdm_apple_configuration_profiles` that don't have a matching row in `host_mdm_apple_profiles`.

since we're going to start `INSERT`ing rows in `host_mdm_apple_profiles` (or at least that's the current plan), we probably need to adjust this to do "EITHER NULL OR NOT NULL" as you describe

Reply:
I think the current implementation is poorly done, having this kind of simple retry logic each time the cron runs sounds like something we want, right?

we probably need to adjust the estimate to fix all of this, that's on me!

Reply:
> having this kind of simple retry logic each time the cron runs sounds like something we want, right?

I think so.

> the current implementation is poorly done

Not at all, I think what's there is a great start and it's normal that as we wire up more features we find some things that need tweaking! I don't think it'd be too complex to have the retry, once I'm back working on #10122 I can address that as part of the PR. I agree that it may need adjusting the estimate a bit, but it's not a huge difference, and it's great to review those complex things (that were implemented when we had less knowledge about how it would all work) as part of the bigger picture now that we know a bit more how we want it to behave. It's not at all on you, those are all things I could've just as easily spotted in code review and didn't! Things weren't that clear then.